src/overlays/ovl_i1/sf_ve1.c

```diff
@@ -1,15 +1,133 @@
 #include "global.h"
 #include "sf_ve1.h"
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i1/sf_ve1/func_i1_801920F0.s")
+extern s32 D_i1_8019C0B8;
+extern s32 D_i1_8019C0BC;
+extern s32 D_i1_8019C0C0;
+
+f32 func_i1_801920F0(f32* arg0, f32 arg1, f32 arg2, f32 arg3, f32 arg4, f32* arg5) {
+    f32 temp_ft4;
+    f32 temp_fv1_2;
+    f32 temp_fv1_3;
+    f32 var_fa1;
+
+    *arg5 = arg1 - *arg0;
+
+    if (*arg5 > 180.0f) {
+        do {
+            *arg5 -= 360.0f;
+        } while (*arg5 > 180.0f);
+    } else if (*arg5 < -180.0f) {
+        do {
+            *arg5 += 360.0f;
+        } while (*arg5 < -180.0f);
+    }
+
+    var_fa1 = *arg5;
+
+    if (var_fa1 != 0.0f) {
+        var_fa1 *= arg2;
+
+        if ((arg4 <= var_fa1) || (temp_ft4 = -arg4, (var_fa1 <= temp_ft4))) {
+            if (arg3 < var_fa1) {
+                var_fa1 = arg3;
+            } else {
+                temp_fv1_2 = -arg3;
+
+                if (temp_fv1_2 > var_fa1) {
+                    var_fa1 = temp_fv1_2;
+                }
+            }
+
+            *arg0 += var_fa1;
+        } else {
+            if (var_fa1 < arg4) {
+                var_fa1 = arg4;
+                *arg0 += arg4;
+
+                if (arg1 < *arg0) {
+                    *arg0 = arg1;
+                }
+            } else if (temp_ft4 < var_fa1) {
+                var_fa1 = temp_ft4;
+                *arg0 += temp_ft4;
+
+                if (*arg0 < arg1) {
+                    *arg0 = arg1;
+                }
+            }
+        }
+    }
+
+    temp_fv1_3 = *arg0;
+
+    if (temp_fv1_3 >= 360.0f) {
+        *arg0 = temp_fv1_3 - 360.0f;
+    } else if (temp_fv1_3 < 0.0f) {
+        *arg0 = temp_fv1_3 + 360.0f;
+    }
+
+    return var_fa1;
+}
+
+void func_i1_801922DC(Object_4C* this) {
+Boss* boss = &gBosses[0];
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i1/sf_ve1/func_i1_801922DC.s")
+    for (i = 0; i < 4; i++, boss++) {
+        if ((boss->obj.status != 0) && (boss->obj.id == OBJ_BOSS_319)) {
+            if (boss->obj.pos.z <= this->obj.pos.z) {
+                D_i1_8019C0B8 = (s32) this->obj.rot.x + 1;
+                this->obj.status = 0;
+            }
+            break;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i1/sf_ve1/func_i1_80192358.s")
+void func_i1_80192358(Object_4C* this) {
+    Boss* boss = gBosses;
```

Comment:
There are still others to fix, please don't use the apply suggestion button, you should do this manually.
```suggestion
Boss* boss = &gBosses[0];